### ISSUE TITLE
delete join-token or redisplay it

### ISSUE THREAD

**Issue by eedgar** [canonical/microceph#132]
There is no way to delete a pending cluster join if it fails.
There is no way to retrieve the join token if you copied it incorrectly.

**Comment by intrepidsilence** [comment]
I am facing this as well.  How should a pending cluster member be removed? I cannot join with the token it gave me and I cannot remove it from the cluster either.

**Comment by masnax** [comment]
The API does support this but there's no CLI command for it in `microceph` it seems. 

For now you can use `microceph cluster sql "delete from internal_token_records"` to remove a join token.

**Comment by intrepidsilence** [comment]
This allowed me to created a new token but that host will still not join. The host is listed in pending status. Is there a way to get rid of the specific host? Why would a token fail to allow a host to join?

**Comment by intrepidsilence** [comment]
Unfortunately, this did not work either:

```
root@zima00:~# microceph cluster list
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
|  NAME  |       ADDRESS       |  ROLE   |                           FINGERPRINT                            |   STATUS    |
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
| lxd1   | 192.168.15.104:7443 | PENDING | f96bed167c08217db2c608a9f2993f7a5a572d71605d98c93cf931aa9a6dd977 | UNREACHABLE |
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
| zima00 | 192.168.15.101:7443 | voter   | b56f4633616dc4373ddc6c7a7ff21e496c52bc55564e4b6c2e9b4e81c20d7f71 | ONLINE      |
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
| zima01 | 192.168.15.102:7443 | voter   | 873d682d33a24a837d395fe508486a2d120a92fbbefd34ab3711257ca4b00dcf | ONLINE      |
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
| zima02 | 192.168.15.103:7443 | voter   | 05b1250369df67f84a882ffdc1235f8062a28e8ba9516104b5a16995d6c51a8f | ONLINE      |
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
root@zima00:~# microceph cluster sql "delete from disks where member_id = (select id from internal_cluster_members where name = 'lxd1')"
Rows affected: 0
root@zima00:~# microceph cluster sql "delete from services where member_id = (select id from internal_cluster_members where name = 'lxd1')"
Rows affected: 0
root@zima00:~# microceph cluster remove lxd1 --force
Error: No remote exists with the given name "lxd1"
root@zima00:~# microceph cluster list
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
|  NAME  |       ADDRESS       |  ROLE   |                           FINGERPRINT                            |   STATUS    |
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
| lxd1   | 192.168.15.104:7443 | PENDING | f96bed167c08217db2c608a9f2993f7a5a572d71605d98c93cf931aa9a6dd977 | UNREACHABLE |
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
| zima00 | 192.168.15.101:7443 | voter   | b56f4633616dc4373ddc6c7a7ff21e496c52bc55564e4b6c2e9b4e81c20d7f71 | ONLINE      |
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
| zima01 | 192.168.15.102:7443 | voter   | 873d682d33a24a837d395fe508486a2d120a92fbbefd34ab3711257ca4b00dcf | ONLINE      |
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
| zima02 | 192.168.15.103:7443 | voter   | 05b1250369df67f84a882ffdc1235f8062a28e8ba9516104b5a16995d6c51a8f | ONLINE      |
+--------+---------------------+---------+------------------------------------------------------------------+-------------+
```

**Comment by intrepidsilence** [comment]
Spoke with masnax on IRC and we kept working on it. Full solution below:

```
microceph cluster sql "delete from internal_token_records"
microceph cluster sql "delete from internal_cluster_members where name = 'lxd1'"
microceph cluster remove lxd1 --force
```

**Comment by sabaini** [comment]
FWIW with PR #155 which fixes issue #49 removing members from microceph should work better

**Comment by UtkarshBhatthere** [comment]
Marking this issue closed.

> Spoke with masnax on IRC and we kept working on it. Full solution below:
> 
> ```
> microceph cluster sql "delete from internal_token_records where token id='<REPLACEME>'"
> microceph cluster sql "delete from internal_cluster_members where name = '<HOSTNAME>'"
> microceph cluster remove <HOSTNAME> --force
> ```
